src/compile/layer.ts

```diff
@@ -104,15 +104,43 @@ export class LayerModel extends Model {
     this.component.layout = parseLayerLayout(this);
   }
 
-  public parseScale() {
+  /**
+   * Convert the domain to an array of data refs. Also, thorw away sorting information
+   * since we always sort the domain when we union two domains.
+   */
+  private static normalizeDomain(domain: DataRefUnionedDomain | FieldRefUnionDomain | VgDataRef): VgDataRef[] {
```

Comment:
I wonder if this method should be here since it will be reused by concat as well.  

This seems like a useful utility.  Maybe it can stay in `scale/domain`?